### PR TITLE
DDF-1766: reorganized DDF Security docs.

### PR DESCRIPTION
**Pull request by ricklarsen**
@oscaritoro @clockard @djblue @vinamartin @stustison @gordocanchola

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/465" target="_blank">ReviewNinja

File: platform/security/docs/src/main/resources/ManContents.adoc [hunk 1]
```diff
@@ -1,120 +1,109 @@
...
+==== Applications Included in {branding} Security
 
-* Security Core
 * Security CAS
+* Security Core
 * Security Encryption
 * Security PEP
 * Security PDP
 * Security STS
+* Security IdP
```
Comment (oscaritoro) [inline platform/security/docs/src/main/resources/ManContents.adoc]: Was this meant to be an alphabetically ordered list? if so IdP should have been after `Security Encryption`

**Comment by oscaritoro** [comment]
There's a couple of instances of DDF in IntContents.adoc that might need changing over to {branding}:
```text
16:43:34-otoro ~/ddf/ddf/platform/security/docs/src (pr-465) $ grep -r "DDF" *
main/resources/IntContents.adoc:. Unzip and install DDF
main/resources/IntContents.adoc:The configuration options are located on the DDF Admin Console under *Opendj Embedded -> Configuration -> LDAP Server*. It currently contains three configuration options.
main/resources/IntContents.adoc:DDF instances can connect to an external LDAP server by installing and configuring the `security-sts-ldaplogin` and `security-sts-ldapclaimshandler` features detailed here.
main/resources/IntContents.adoc:The DDF Security STS Service performs authentication of a user by delegating the authentication request to an STS. This is different than the services located within the Security PDP application as those ones only perform authorization and not authentication.
main/resources/IntContents.adoc:This bundle is installed by default and is required for DDF to operate.
main/resources/IntContents.adoc:The DDF Security PDP Java Realm exposes a realm service that makes decisions on authorization requests using the attributes stored within the metacard to determine if access should be granted. 
```

File: platform/security/docs/src/main/resources/ManContents.adoc (hunk 1 above)
Comment (ricklarsen) [inline platform/security/docs/src/main/resources/ManContents.adoc]: I think IdP was added at the end because it was added last. Just made it alphabetical.

**Comment by ricklarsen** [comment]
fixed.

**Comment by vinamartin** [comment]
:shipit: LGTM

![Ship it]
(https://encrypted-tbn1.gstatic.com/images?q=tbn:ANd9GcTSISWfDcEZanSMYi6SfhH53xVQUDtmnyLOUGOgLCYyZ6IfGh9Ssg)

**Comment by oscaritoro** [comment]
Ship it! :+1:

**Comment by gordocanchola** [comment]
:+1:

File: platform/security/docs/src/main/resources/ExtContents.adoc [hunk 2]
```diff
@@ -1,20 +1,78 @@
...
+The `TokenValidator` CXF interface must be implemented by any custom token validator class.
+The `canHandleToken` and `validateToken` methods must be overridden.
+The `canHandleToken` method should return true or false based on the `ValueType` value of the token that the validator is associated with.
+The validator may be able to handle any number of different tokens that you specify.
+The `validateToken` method returns a `TokenValidatorResponse` object that contains the `Principal` of the identity being validated and also validates the `ReceivedToken` object that was collected from the RST (`RequestSecurityToken`) message.
 
-At the moment, validators must be added to the cxf-sts.xml blueprint file manually. There is a section labeled as the "Delegate Configuration" in that file. That is where the validator must be added along with the existing validators. In the future, we expect this to be pluggable without changes to the blueprint file.
+==== Using
+
+At the moment, validators must be added to the `cxf-sts.xml` blueprint file manually.
```
Comment (stustison) [inline platform/security/docs/src/main/resources/ExtContents.adoc]: This isn't true. Validators are picked up as OSGi services now.

Comment (ricklarsen) [inline platform/security/docs/src/main/resources/ExtContents.adoc]: fixed.

**Comment by oscaritoro** [comment]
Latest update looks good :+1:

**Pull request merged**